pkg/apis/kops/validation/instancegroup_test.go

```diff
@@ -294,3 +298,65 @@ func TestIGCloudLabelIsIGName(t *testing.T) {
 		testErrors(t, g.label, errs, g.expected)
 	}
 }
+
+func TestInstanceMetadataOptions(t *testing.T) {
+	tests := []struct {
+		ig       *kops.InstanceGroup
+		cloud    fi.Cloud
+		expected []string
+	}{
+		{
+			ig: &kops.InstanceGroup{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "some-ig",
+				},
+				Spec: kops.InstanceGroupSpec{
+					Role: "Node",
+					InstanceMetadata: &kops.InstanceMetadataOptions{
+						HTTPPutResponseHopLimit: fi.Int64(1),
+						HTTPTokens:              fi.String("optional"),
+					},
+				},
+			},
+			cloud:    gce.InstallMockGCECloud("us-test1", "testproject"),
+			expected: []string{"Invalid value::spec.instanceMetadata"},
+		},
+		{
+			ig: &kops.InstanceGroup{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "some-ig",
+				},
+				Spec: kops.InstanceGroupSpec{
+					Role: "Node",
+					InstanceMetadata: &kops.InstanceMetadataOptions{
+						HTTPPutResponseHopLimit: fi.Int64(1),
+						HTTPTokens:              fi.String("abc"),
+					},
+				},
+			},
+			cloud:    awsup.BuildMockAWSCloud("us-east-1", "abc"),
+			expected: []string{"Invalid value::spec.instanceMetadata.httpTokens"},
+		},
+		{
+			ig: &kops.InstanceGroup{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "some-ig",
+				},
+				Spec: kops.InstanceGroupSpec{
+					Role: "Node",
+					InstanceMetadata: &kops.InstanceMetadataOptions{
+						HTTPPutResponseHopLimit: fi.Int64(-1),
+						HTTPTokens:              fi.String("required"),
+					},
+				},
+			},
+			cloud:    awsup.BuildMockAWSCloud("us-east-1", "abc"),
+			expected: []string{"Invalid value::spec.instanceMetadata.httpPutResponseLimit"},
```

Comment:
```suggestion
			expected: []string{"Invalid value::spec.instanceMetadata.httpPutResponseHopLimit"},
```

Reply:
Thank you for these suggestions @hakman  :)

Reply:
Hope they help. :)